TestResultSummaryService/parsers/Tap.js

```diff
@@ -0,0 +1,347 @@
+const Parser = require('./Parser');
+const Utils = require(`./Utils`);
+const decompress = require("decompress");
+const ObjectID = require( 'mongodb' ).ObjectID;
+const path = require('path');
+const { TestResultsDB, OutputDB } = require('../Database');
+
+let filePath = null
+let outputDb = null;
+let testResultDb = null
+let timestamp = Date.now()
+class Tap extends Parser {
+    static canParse(filePath) {
+        let path = require('path')
+        if (filePath) {
+            if (path.extname(filePath) == '.zip') {
+                return true;
+            } 
+            else {
+                return false
+            }
+        } else {
+            return false;
+        }
+    }
+
+    static async parse(zipPath) {   
+        const files = await decompress(zipPath,  "dist"); // unzipping
+        const zipName = path.basename(zipPath).replace(".zip", "")
+        filePath = path.join(__dirname, '../dist/' + zipName) 

```

Comment:
filePath is only used in `getTapFileTests` which is called at the very end and I wanted to avoid passing filePath all the way down the chain. I did change it so that zipPath (now renamed zip) would become the global variable since technically there's no need to create zipName or filePath at the very beginning